### PR TITLE
[MRESOLVER-540] TransferResource can reveal more about resource

### PR DESCRIPTION
**Pull request by cstamas**
Provide extra information about the resource being transferred.

---

https://issues.apache.org/jira/browse/MRESOLVER-540

### CODE REVIEW

**Comment by cstamas** [comment]
Main inspiration is to make stunts like these not required: https://github.com/apache/maven-mvnd/pull/930

File: maven-resolver-api/src/main/java/org/eclipse/aether/transfer/TransferResource.java [hunk 1]
```diff
@@ -23,17 +23,43 @@
...
  * Describes a resource being uploaded or downloaded by the repository system.
  */
 public final class TransferResource {
 
+    /**
+     * The type of this resource.
+     *
+     * @since 2.0.0
+     */
+    public enum ResourceType {
```
Comment (gnodet) [inline maven-resolver-api/src/main/java/org/eclipse/aether/transfer/TransferResource.java:38]: I wonder if that one is really needed, as one could use `instanceof` on the resource (and a cast will be needed anyway)...

**Review by gnodet — APPROVED** [review]
Not sure about the enum...

File: maven-resolver-api/src/main/java/org/eclipse/aether/transfer/TransferResource.java (hunk 1 above)
Comment (cstamas) [reply maven-resolver-api/src/main/java/org/eclipse/aether/transfer/TransferResource.java:38]: I kinda agree with you, moreover as these events are emitted ONLY by connector. For example, Transport API of Maven4 completely circumvents connector, and goes directly to transport... so yeah, due this (connector) type can really only be Artifact or Metadata

**Comment by cstamas** [comment]
Will remove/simplify it

**Comment by cstamas** [comment]
Simplified, removed cruft and is not simple and nice. @gnodet pls review

**Pull request merged**